src/components/Button/Button.tsx

```diff
@@ -0,0 +1,66 @@
+import { ReactNode, ButtonHTMLAttributes } from 'react';
+import classNames from 'classnames/bind';
+import styles from './Button.module.scss';
+
+const cn = classNames.bind(styles);
+
+const BUTTON_COLOR = {
+  BACKGROUND_GRAY_40: 'background-gray-40',
+  BACKGROUND_PRIMARY: 'background-primary',
+  BACKGROUND_PRIMARY_60: 'background-primary-60',
+  OUTLINE_GRAY_40: 'outline-gray-40',
+  OUTLINE_PRIMARY: 'outline-primary',
+  OUTLINE_PRIMARY_60: 'outline-primary-60',
+} as const;
+
+type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];
+
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  backgroundColor?: ButtonColorType;
+  hoverColor?: ButtonColorType;
+  radius?: 4 | 8;
+  width?: 'parent-full' | 72 | 90 | 120 | 154 | 320;
+  fontSize?: 14 | 18 | 20 | 24;
+  paddingVertical?: 8 | 20;
+  children: ReactNode;
+}
+
+/**
+ * button component documentation
+ * @param {ButtonColorType} [props.backgroundColor] - 버튼에 적용할 디자인
+ * @param {ButtonColorType} [props.hoverColor] - 버튼에 hover 시 적용할 디자인
+ * @param {number} [props.redius] - 버튼의 border-radius 크기 / 4px or 8px
+ * @param {number} [props.width] - 너비의 크기 px 단위 / 'parent-full'은 100%
+ * @param {number} [props.fontSize] - 폰트 크기 px 단위
+ * @param {number} [props.paddingVertical] - 수직 방향 패딩 값
+ * @param {funtion} [props.onClick] - 버튼 클릭 시 실행할 함수
+ * @param {JSX.Element} [props.children] - 버튼 안에 들어갈 내용
+ * @returns {JSX.Element}
+ */
+
+export default function Button({
+  backgroundColor = BUTTON_COLOR.BACKGROUND_PRIMARY,
+  hoverColor,
+  radius = 8,
+  width = 'parent-full',
+  fontSize = 18,
+  paddingVertical = 20,
+  children,
+  ...rest
+}: ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>): JSX.Element {
+  const widthClassName = width === 'parent-full' ? 'parent-full' : `width-${width}`;
```

Comment:
parent-full이 없으면 이런 식으로 하면 될 것 같습니다
```suggestion
  const widthClassName = width ? `width-${width}`: 'parent-full';
```

Reply:
오홍!!! 그렇네여!! 바아로 수정 반영하겠습니닷~~!!!